code/clue.js

```diff
@@ -1,59 +1,396 @@
+const pickWeapon = () => {
+  if (killerIsPicked && !weaponIsPicked) {
+    document.getElementById('weaponLoader').style.visibility='visible'
+    setTimeout(function () {
+      document.getElementById('weaponLoader').style.visibility='hidden'
+      mystery.weapon = randomSelector(weapons); //Randomly select a weapon from weapons and add it to the mystery object
+      document.getElementById('weaponCard').style.background = mystery.weapon.color
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`
+  document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight}`
+  document.getElementById('weaponFlexibility').innerHTML = `Flexibility: ${mystery.weapon.flexibility}`
+  document.getElementById('weaponImage').src = `${mystery.weapon.image}`
+  weaponIsPicked = true
+    }, 2500)
+  }
+};
+document.getElementById('weaponCard').addEventListener('click', pickWeapon);
+
+//If pickKiller AND pickWeapon function is excecuted/cards are clicked then excecute this function on the pickroom:
+const pickRoom = () => {
+  if (killerIsPicked && weaponIsPicked) {
+    document.getElementById('roomLoader').style.visibility='visible'
+    document.getElementById('revealButton').style.visibility='visible'//Makes the reveal-button visible when the room-card has been clicked.
+    setTimeout(function () {
+      document.getElementById('roomLoader').style.visibility='hidden'
+      mystery.room = randomSelector(rooms);//Randomly select a room from rooms and add it to the mystery object
+      document.getElementById('roomName').innerHTML = `${mystery.room.name}`
+      document.getElementById('roomImage').src = `${mystery.room.image}`
+      roomIsPicked = true
+    }, 2500)
+  }
+};
+document.getElementById('roomCard').addEventListener('click', pickRoom); 
+
+// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. 
+const revealMystery = (mystery) => {
+  //Invoke when button is clicked and show elements from mystery object in text:
+document.getElementById('mystery').innerHTML = `The murder was committed by ${mystery.killer.firstName} in the ${mystery.room.name} with ${mystery.weapon.name}` 
+document.getElementById('revealButton').innerHTML = `PLAY AGAIN?` //change the text on button.
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+//BELOW I TRIED TO MAKE A FUNCTION SO THAT THE PAGE RELOADS WHEN 'PLAY AGAIN'-BUTTON IS CLICKED. IT DIDN'T WORK. I wanted to leave this here just to show the code-reviewer what I tried and maybe get pointers on how to achieve this.
+
+//THIS DOESN'T WORK:
+//  const playAgain = () => {
+//    if (document.getElementById('revealButton').innerHtml === 'PLAY AGAIN?') {
+//         //location.reload();
+//         revealButton().onclick === pageReload; 
+//         //revealButton().onclick === location.reload();
+//     } 
+//   };
+//document.getElementById('revealButton').addEventListener('click', playAgain);
```

Comment:
I can't see exactly why this isn't working, but I have noticed that you haven't changed killerIsPicked, roomIsPicked, weaponIsPicked variables back to false.

Reply:
I made it work now with help from Van and Max :)